### DISCUSSION TITLE
How to disable Equinox internal LoggerFactory

### DISCUSSION THREAD

**Discussion by martin-petzold** [eclipse-equinox/equinox#593] [General]
I am using Eclipse Equinox as my OSGi framework implementation. I have my own launcher implementation. Now I would like to use also my own log implementation (LoggerFactory) exclusively.

However, after launch I always find the org.eclipse.osgi LoggerFactory implementation. This is in conflict with my implementation. So currently all log calls reach Equinox implementation but not my implementation.

I have a very performance critical application. Using LogReader is NOT an option!

How can I disable Equinox internal LoggerFactory implementation?

**Comment by martin-petzold** [comment]
I found a work-around: Register your own LoggerFactory with (highest) service ranking:

```
Dictionary<String, Object> properties = new Hashtable<String, Object>();
		
properties.put(Constants.SERVICE_RANKING, Integer.MAX_VALUE);
		
this.serviceRegistration = bundleContext.registerService(new String[] { LogService.class.getName(), LoggerFactory.class.getName() }, this, properties);
```

**Reply by laeubi** [reply]
> I found a work-around: Register your own LoggerFactory with (highest) service ranking:

This actually is not a workaround but how it is supposed to override services provided by other bundles.
